docs/Build.md

```diff
@@ -10,41 +10,67 @@ The design of the build system has the following main goals:
 
 * works identically on every platform
 * has minimal run-time dependencies:
-    - a working installation of `stack`
+    - `stack`
     - `git`
-* is completely functional right after simple `git clone`
-* one-stop-shop for building all executables required for using `hie` in IDEs.
+* is completely functional right after a simple `git clone` and after every `git pull`
+* one-stop-shop for building and naming all executables required for using `hie` in IDEs.
+* fails with meaningful error messages in known circumstances
```

Comment:
* prevents certain build failures by either identifying a failed precondition (such as wrong `stack` version) or by performing the necessary steps so users can't forget them (such as invoking `git` to update submodules)

Reply:
I will include this wording